### PR TITLE
specs out the additional fields for nmve_ssd

### PR DESCRIPTION
**Pull request by geojaz**
Adds `local_nvme_ssd_block` to `node_config` block in the `google_container_node_pool`.

Also I needed to update the go.sum and go.mod and wasn't really clear on the right process.
Does this look right? Is there a "right" way to do it? Mine was really hacky :)

Closes https://github.com/hashicorp/terraform-provider-google/issues/13829

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**
```release-note:enhancement
container: added `local_nvme_ssd_block` to `node_config` block in the `google_container_node_pool`.
```

### CODE REVIEW

**Comment by geojaz** [comment]
I can't see the logs for the failures, but i think it might have something to do with updating the deps for `google.golang.org/api v0.111.0`.
I'll have to take a closer look to the tests on my side unless you can share some additional details- feel free to ping me on gchat @ehole

**Comment by geojaz** [comment]
oops..

**Comment by shuyama1** [comment]
Tests failed due to:
```
go: bitbucket.org/creachadair/stringset@v0.0.8: missing go.sum entry; to add it:
	go mod download bitbucket.org/creachadair/stringset
make: *** [vet] Error 1
GNUmakefile:26: recipe for target 'vet' failed
make: *** No rule to make target 'tools'.  Stop.
==> Checking that code complies with gofmt requirements...
go vet
go: bitbucket.org/creachadair/stringset@v0.0.8: missing go.sum entry; to add it:
	go mod download bitbucket.org/creachadair/stringset
make: *** [vet] Error 1
GNUmakefile:26: recipe for target 'vet' failed
==> Checking that code complies with gofmt requirements...
go vet
go: bitbucket.org/creachadair/stringset@v0.0.8: missing go.sum entry; to add it:
	go mod download bitbucket.org/creachadair/stringset
make: *** [vet] Error 1
```

**Comment by geojaz** [comment]
I think this works finally, sorry for all the iterations. It's such a slow feedback loop and switching between the provider and magic modules was at least for me introducing errors 🤦 
```
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccContainerNodePool_localNvmeSsdBlockConfig -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccContainerNodePool_localNvmeSsdBlockConfig
=== PAUSE TestAccContainerNodePool_localNvmeSsdBlockConfig
=== CONT  TestAccContainerNodePool_localNvmeSsdBlockConfig
--- PASS: TestAccContainerNodePool_localNvmeSsdBlockConfig (727.63s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta 728.579s
```

**Review by shuyama1 — COMMENTED** [review]
LGTM! only some small comments. Plus, any reason we add in beta provider exclusively? I do see this field is supported in V1 API though.

**Comment by geojaz** [comment]
> LGTM! only some small comments. Plus, any reason we add in beta provider exclusively? I do see this field is supported in V1 API though.
Good call- I'll push it to GA. I effectively just copied the implementation of `ephemeral_storage_config`- looks like this should also graduate... I won't include that here.

**Comment by geojaz** [comment]
Pushed with your suggestions, I couldn't get the tests on generated GA provider to run on my system- there were lots of seemingly unrelated errors. Not sure what's up

File: mmv1/third_party/terraform/utils/node_config.go.erb [hunk 1]
```diff
@@ -589,6 +608,14 @@ func expandNodeConfig(v interface{}) *container.NodeConfig {
...
 	}
 <% end -%>
+<% unless version == 'ga' -%>
+	if v, ok := nodeConfig["local_nvme_ssd_block_config"]; ok && len(v.([]interface{})) > 0 {
+		conf := v.([]interface{})[0].(map[string]interface{})
+		nc.LocalNvmeSsdBlockConfig = &container.LocalNvmeSsdBlockConfig{
+			LocalSsdCount: int64(conf["local_ssd_count"].(int)),
+		}
+	}
+<% end -%>
```
Comment (shuyama1) [inline mmv1/third_party/terraform/utils/node_config.go.erb:618]:
```suggestion
	if v, ok := nodeConfig["local_nvme_ssd_block_config"]; ok && len(v.([]interface{})) > 0 {
		conf := v.([]interface{})[0].(map[string]interface{})
		nc.LocalNvmeSsdBlockConfig = &container.LocalNvmeSsdBlockConfig{
			LocalSsdCount: int64(conf["local_ssd_count"].(int)),
		}
	}
```
we also need to remove the flag here

**Review by shuyama1 — CHANGES_REQUESTED** [review]
GA test failed on TC: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstream/383054
It's probably due to one of the beta flags not removed

**Comment by shuyama1** [comment]
GA test passed: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloud_ProviderGoogleCloudMmUpstream/383064

**Review by shuyama1 — APPROVED** [review]
LGTM!

**Pull request merged**